src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -2310,6 +2310,26 @@ void processContextElement
                     getEntitiesCollectionName(tenant).c_str(),
                     query.toString().c_str()));
 
+    std::map<std::string, std::string>::iterator it = uriParams.find("op");
+
+    // if is an append and uri has not possibility of options
+    if (strcasecmp(action.c_str(), "append") == 0 && it == uriParams.end())
+    {
+        EntityIdVector vec;
+        vec.push_back(enP);
+
+        AttributeList attList;
+        Restriction res;
+        ContextElementResponseVector rsp;
+        std::string err;
+        entitiesQuery(vec, attList, res, &rsp, &err, true, tenant, servicePathV);
+
+        if (rsp.size() && !rsp.get(0)->prune)
+        {
+            throw OrionCreateException("Entity already exists");
```

Comment:
Not that I am especially *against* exceptions, even though it is an extremely complicated method to simply return an error code, but in the source code of the broker we don't throw exceptions. We do catch them, as we have no other choice as e.g. the mongo client library throws exceptions.

The source code of orion uses return values and checks for the return value after each call - the simpler and faster 'C method' ...

You will soon understand that the broker is implemented in C, with very few features from C++.
In my personal opinion, the less C++, the better ...

I would prefer to uniformally keep using return values and avoid throwing exceptions.
Later we can discuss and perhaps change this choice of implementation method but until then ...